packages/smooth_app/lib/pages/product/product_image_gallery_view.dart

```diff
@@ -116,17 +116,17 @@ class _ProductImageGalleryViewState extends State<ProductImageGalleryView> {
                 slivers: <Widget>[
                   _buildTitle(appLocalizations.selected_images, theme: theme),
                   SmoothImagesSliverList(
-                    imagesData: selectedImages,
+                    imagesData: _selectedImages,
                     onTap: (ProductImageData data, _) => data.imageUrl != null
-                        ? _openImage(data, product)
-                        : _newImage(data, product),
+                        ? _openImage(data, product.barcode!)
```

Comment:
I meant, you don't have to pass a product/barcode parameter _at all_.
You can just use `widget.product.barcode!` _inside_ your methods.